utility/check_baseline.py

```diff
@@ -0,0 +1,202 @@
+    ants[61]=katpoint.Antenna('m061, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  -16.409 -2323.779 21.304')
+    ants[62]=katpoint.Antenna('m062, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  -1440.632 -2503.773  21.683')
+    ants[63]=katpoint.Antenna('m063, -30:42:39.8, 21:26:38.0, 1086.6, 13.5,  -3419.585 -1840.480 16.383')
+    return ants
+
+
+def get_uvw(target,ants,timevalues):
+    # Set up empty uvw then loop through the half the antennas (n,m)
+    # and calculate u,v,w for the target and antennas n and m
+    u,v,w=[],[],[]
+    for n in range(np.shape(ants)[0]-1):
+        for m in range(n+1,np.shape(ants)[0]):
+            u_new,v_new,w_new=target.uvw(ants[n], timevalues ,ants[m])
+            u.extend(u_new)
+            v.extend(v_new)
+            w.extend(w_new)
+    return np.array(u),np.array(v),np.array(w)
+    #u,v,w = get_uvw(target, ants, katpoint.Timestamp(str(time1)).secs + np.linspace(0.0, upt2 * 60. * 60., 360))
+
+def get_xyz(ants):
+    # Set up empty xyz then loop through the half the antennas (n,m)   
+    baselineno = (np.shape(ants)[0]* (np.shape(ants)[0]-1))//2
+    x,y,z=np.zeros((baselineno)),np.zeros((baselineno)),np.zeros((baselineno))
+    i=0
+    for n in range(np.shape(ants)[0]-1):
+        for m in range(n+1,np.shape(ants)[0]):
+            x[i],y[i],z[i]= np.array(ants[n].position_enu)-np.array(ants[m].position_enu)
+            i = i+ 1
+    return x,y,z
+
+def verbosef(string,verbosebool):
+    if verbosebool :
+        print(string)
+        
+def test_array(ants,reduced_ants,minfrac=None,mintotal=None,verbose=False,bins =None ):
+    observation_good = True
+    x,y,z = get_xyz(ants)
+    x1,y1,z1 = get_xyz(reduced_ants)
+    verbosef("There are %i antennas."%(len(reduced_ants)),verbose)
+    if mintotal is not None:
+        if len(reduced_ants) < mintotal:
+            observation_good = False
+            verbosef("There are not enough antennas. Only %i antennas avalible."%(len(reduced_ants)),verbose)
+            verbosef("No fewer than %i antennas."%(mintotal),verbose)
+    else:
+        verbosef("NO constraints on the number of antennas",verbose)
+
+    if bins is not None:
+        total = np.histogram((np.sqrt(x**2+y**2)),bins=bins)[0]
+        actual= np.histogram((np.sqrt(x1**2+y1**2)),bins=bins)[0]
+        if not np.all(actual/total >= minfrac):
+            observation_good = False
+        verbosef('# Fraction of available baselines as a function of baseline length:',verbose) 
+        for i in range(len(bins)-1):
+            passfail = "---FAIL---"
+            if (actual/total)[i] >= minfrac[i] :
+                passfail = "    ok     "
+            output_str = "# %4d - %4d m:   %0.3f %% available  >=   %0.3f %% needed -- %s"%(bins[i],bins[i+1],(actual/total)[i],minfrac[i],passfail)
+            verbosef(output_str,verbose)  
+    return observation_good
+
+class projects_contraints(): # I suspect that this is a horrible misuse of a class
```

Comment:
Not too horrible :-)

There is also `namedtuple` for this. In other words:
```Python
from collections import namedtuple
ProjectConstraints = namedtuple('ProjectConstraints', 'minfrac bins mintotal')

projects['MHONGOOSE'] = ProjectConstraints(
    bins=np.array([0, 50, 100, 200, 400, 1000, 3000, 6000, 9000]),
    minfrac=np.array([0.80, 0.66, 0.66, 0.66, 0.66, 0.66, 0.50, 0.00]),
    mintotal = 58
)
```

Reply:
done